File: benchmarks/fixture/java/SessionService.java
package fixture;

public final class SessionService {
    public record Credentials(String accessToken, long expiresAt) {}

    public static Credentials renewSessionCredentials(String refreshToken, long now) {
        if (refreshToken == null || refreshToken.isBlank()) {
            throw new IllegalArgumentException("refresh token required");
        }
        return new Credentials("access:" + refreshToken, now + 900);
    }

    public static boolean validateAccessToken(long expiresAt, long now) {
        return expiresAt > now;
    }
}

File: benchmarks/fixture/csharp/SessionService.cs
namespace Fixture;

public static class SessionService
{
    public static (string AccessToken, long ExpiresAt) RenewSessionCredentials(string refreshToken, long now)
    {
        if (string.IsNullOrWhiteSpace(refreshToken)) throw new ArgumentException("refresh token required");
        return ($"access:{refreshToken}", now + 900);
    }

    public static bool ValidateAccessToken(long expiresAt, long now) => expiresAt > now;
}
